src/libstd/build.rs

```diff
@@ -46,6 +46,8 @@ fn main() {
     } else if target.contains("dragonfly") || target.contains("bitrig") ||
               target.contains("netbsd") || target.contains("openbsd") {
         println!("cargo:rustc-link-lib=pthread");
+    } else if target.contains("solaris") {
+        println!("cargo:rustc-link-lib=gcc_s");
```

Comment:
I believe if this is linked through libunwind then we can avoid linking it here as well